### DISCUSSION TITLE
Move EMODTask sweeping methods into iTask OR a mid-level Task

### DISCUSSION THREAD

**Discussion by ckirkman-IDM** [InstituteforDiseaseModeling/idmtools#1889] [General]
The new Light EMOD (LEMOD) python model (which is not EMOD) uses the following call:

```
from emodpy.emod_task import EMODTask
...
EMODTask.set_parameter_sweep_callback
```
... which requires having emodpy installed to load. This is not elegant.

I propose we move set_parameter_sweep_callback and related parameter read/write calls that are not EMOD-specific to an intermediate Task class, e.g. ModelTask, so that non-EMOD models do not need emodpy installed/loaded.

Setting the estimate to 1 assuming that EMODTask has been pretty well vetted and this is largely a cut-and-paste job. Any complications should come from deciding if any new tests are needed or if they should be jumbled around, too.

**Comment by devclinton** [comment]
@ckirkman-IDM  Good idea. We can't do ITask because it assumes no config. We might be able to make an abstract version of this method between though. Like ISweepableTask or something that has to implement said method. That at least allows us at the core to stay oblivious to config at the ITask level.

For other models, I think we need better documentation. There are a few routes to go. 

1. In memory config stored on python object. This is my go to. I usually attach a dict object on my task and then use callbacks to define my sweep. It could be made more fluid. @devclinton - Note example of this later to this ticket
2. JSON Config. There is a base model in the idmtools-models package that provides the method for config. A user would just need to subclass to inform how to run. We have derived versions for python and R, but maybe we just need a JSONConfiguredCommandTask or the base configured task
3. Fully derive the model.

For your suggest, we could add that ISweepableTask to JSONConfiguredBaseTask and then also EMODTask.

**Comment by ZDu-IDM** [comment]
@devclinton @ckirkman-IDM Current idmtools-Calibra doesn't have map_replicates_callback and CalibManager has itself own-defined run_number method. Therefore there is no dependency on EMODTask (EMODTask.set_parameter_sweep_callback) at all.

We don't need to change anything.
